build-tools/packages/build-tools/src/type-test-generator/compatibility.ts

```diff
@@ -163,3 +162,37 @@ namespace Test7 {
 	// but only in one direction:
 	type _check5 = requireAssignableTo<TypeOnly<Test7.Baz>, TypeOnly<Test6.Foo>>;
 }
+
+namespace Test_TypeOnly_Preserves_Primitives {
+	// The 'null' and 'undefined' values cannot be nominal typed (they are not
+	// valid enum values and intersection ('&') with 'null'/'undefined' -> 'never').
+	// Just verify that the 'null' and 'undefined' values are preserved.
+	type _check_undefined1 = requireAssignableTo<TypeOnly<undefined>, undefined>;
+	type _check_undefined2 = requireAssignableTo<undefined, TypeOnly<undefined>>;
+
+	type _check_null1 = requireAssignableTo<TypeOnly<null>, null>;
+	type _check_null2 = requireAssignableTo<null, TypeOnly<null>>;
+
+	// Types that extend 'number' and 'string' are stripped to their primitive types.
+	// (Nominal typing is erased.)
+	type brandedNumber = number & { brand: "Number" };
+	type _check_number1 = requireAssignableTo<TypeOnly<brandedNumber>, number>;
+	type _check_number2 = requireAssignableTo<number, TypeOnly<brandedNumber>>;
```

Comment:
this branding isn't nominal typing: its actually structural. Two different declarations of this same type (ex: in two versions of the same package) are assignable to eachother.

Thus ideally TypeOnly would preserve these brands. The fact that this type brand is getting lost is a limitation of TypeOnly, and not a feature we want to ensure doesn't break: ideally, we would preserve the brand.

The above case, _check_number1 ensures TypeOnly doesn't forget the value is a number: that is good.

```suggestion
   // Due to limitations of the current version of TypeOnly, brands on primitives are lost:
	type _check_number2 = requireAssignableTo<number, TypeOnly<brandedNumber>>;
```